app/search/forms.py

```diff
@@ -40,16 +55,97 @@ def add_fields(self):
             ),
             FieldsConstant.Q: CharField(),
+            FieldsConstant.RECORD_DATE_FROM: DateComponentField(
+                label="Record date from",
+                is_from_date=True,
+                required=False,
+            ),
+            FieldsConstant.RECORD_DATE_TO: DateComponentField(
+                label="Record date to",
+                is_from_date=False,
+                required=False,
+            ),
+        }
+
+    @abstractmethod
+    def get_specific_fields(self):
+        """Returns fields specific to the form subclass"""
+        pass
+
+    def add_fields(self):
+        """Combines common and specific fields - called by BaseForm.__init__"""
+        fields = self.get_common_fields()
+        fields.update(self.get_specific_fields())
+        return fields
+
+    def cross_validate(self) -> list[str]:
+        """Validate date ranges - from dates should not be later than to dates"""
+        errors = []
+
+        # Validate record date range
+        rd_from = self.fields[FieldsConstant.RECORD_DATE_FROM].cleaned
+        rd_to = self.fields[FieldsConstant.RECORD_DATE_TO].cleaned
+
+        if rd_from and rd_to and rd_from > rd_to:
+            errors.append("Record date 'from' cannot be later than 'to' date")
+
+        # Validate opening date range (only for TNA forms that have these fields)
+        if (
+            FieldsConstant.OPENING_DATE_FROM in self.fields
+            and FieldsConstant.OPENING_DATE_TO in self.fields
+        ):
+
+            od_from = self.fields[FieldsConstant.OPENING_DATE_FROM].cleaned
+            od_to = self.fields[FieldsConstant.OPENING_DATE_TO].cleaned
+
+            if od_from and od_to and od_from > od_to:
+                errors.append(
+                    "Opening date 'from' cannot be later than 'to' date"
+                )
+
+        return errors
+
+    def get_api_date_params(self) -> list[str]:
+        """Get formatted date parameters as filter list for API request"""
+        date_filters = []
+
+        def get_field_value(field_name):
+            if field_name in self.fields and hasattr(
+                self.fields[field_name], "format_for_api"
+            ):
+                return self.fields[field_name].format_for_api()
+            return None
+
+        # Record dates (common to all forms)
+        if rd_from := get_field_value(FieldsConstant.RECORD_DATE_FROM):
+            date_filters.append(f"coveringFromDate:(>={rd_from})")
+        if rd_to := get_field_value(FieldsConstant.RECORD_DATE_TO):
+            date_filters.append(f"coveringToDate:(<={rd_to})")
+
+        # Opening dates (only for TNA forms - check if fields exist)
+        if od_from := get_field_value(FieldsConstant.OPENING_DATE_FROM):
+            date_filters.append(f"openingFromDate:(>={od_from})")
+        if od_to := get_field_value(FieldsConstant.OPENING_DATE_TO):
+            date_filters.append(f"openingToDate:(<={od_to})")
+
+        return date_filters
```

Comment:
Rather than processing this at the form level, it might be more appropriate to handle it directly through the `APIMixin` here -> https://github.com/nationalarchives/ds-catalogue/blob/243390422b7b6ef600882fd2a719915d895e6a34/app/search/views.py#L59